pkg/test/connector.go

```diff
@@ -88,21 +88,36 @@ func (suite *ConnectorTestSuite) TestAll() {
 					for _, p := range planRes.Msg.GetPartitions() {
 						var cursor []byte
 						var count int
-						for {
-							res, err := c.ListData(ctx, connect.NewRequest(&adiomv1.ListDataRequest{
+						var numLoops = 5
```

Comment:
hmm this is a bit confusing to me. I think we should be running to completion as before, we just want to assert that we have paginated through at least X pages.

And then for counting, we should only be counting res1, even though we are doing it twice with res2 and comparing they have the same data each time.